slick/src/main/scala/slick/lifted/Query.scala

```diff
@@ -54,6 +54,16 @@ sealed abstract class Query[+E, U, C[_]] extends QueryBase[C[U]] { self =>
   def filterNot[T <: Rep[_]](f: E => T)(implicit wt: CanBeQueryCondition[T]): Query[E, U, C] =
     filterHelper(f, node => Library.Not.typed(node.nodeType, node) )
 
+  /** Applies the given predicate, if the optional value is present,
+    * if the value is not present, the predicate will not be part of the query. */
+  def filterOpt[V, T <: Rep[_] : CanBeQueryCondition](optValue: Option[V])(f: (E, V) => T): Query[E, U, C] =
+    optValue.map(v => withFilter(a => f(a, v))).getOrElse(this)
+
+  /** Applies the given predicate, if the given boolean parameter evaluates to true,
+    * if the value is not present, the predicate will not be part of the query. */
```

Comment:
Probably meant `if the value evaluates to false` - "not present" doesn't really match a boolean (I assume it's a copy-paste issue)

Reply:
:+1: I'll make the change.